synaptic_reconstruction/training/semisupervised_training.py

```diff
@@ -61,8 +61,12 @@ def get_unsupervised_loader(
     else:
         roi = None
 
+    z,y,x = patch_shape
+    ndim = 2 if z == 1 else 3
+    print("ndim is: ", ndim)
```

Comment:
I think you also need to pass the `ndim` argument to the `RawDataset` a few lines below.